### DISCUSSION TITLE
How does OpenLane use LEF file information to place and route designs

### DISCUSSION THREAD

**Discussion by jarvis-lorry** [The-OpenROAD-Project/OpenLane#1034] [General]
In the sky130A LEF file, for the cell macros, I can see interconnect (li1) layers for pins and power/ground, but the other layers and connections in the final layout aren’t included there.  For example, the li1 drawing layer and diffusion layers are added later.  Can anyone tell me how OpenLane gets the extra information needed to draw in the cells?

**Comment by armleo** [comment]
The GDS/MAGs contains everything. LEFs are abstract representation of these cells. OpenLane needs only the layers that can connect between multiple cells (metals ,vias, nwell, pwell).
